src/gt4py/next/program_processors/runners/dace_iterator/itir_to_sdfg.py

```diff
@@ -189,6 +197,103 @@ def add_storage(
             raise NotImplementedError()
         self.storage_types[name] = type_
 
+    def generate_temporaries(
```

Comment:
I am not sure about the name.
What confuses me is that `self.tmps` are the temporaries, or not?
And later you call this function only if `self.tmps` is not empty, which seems a bit contradictory.